File: examples/handle_recs.pl
#<![CDATA[
use strict;
use RASCH;

# initialize libRASCH
my $ra = new RASCH or die "error initializing libRASCH\n";

# open measurement
my $meas = $ra->open_meas($ARGV[0], '', 0) or
    die "can't open measurement $ARGV[0]\n";

# get root recording
my $rec = $meas->get_first_rec(0) or
    die "can't get root recording\n";

# get some infos about recording
my $v = $rec->get_info(info => 'rec_num_devices');
my $num_dev = $v->value();
$v = $rec->get_info(info => 'rec_num_channel');
my $num_ch = $v->value();
$v = $rec->get_info(info => 'rec_name');
my $rec_name = $v->value();
$v = $rec->get_info(info => 'rec_date');
my $rec_date = $v->value;
print "measurement $rec_name\nrecorded at $rec_date\n" .
    "#devices=$num_dev\n#channels=$num_ch\n\n";

# print name for every device
print "infos about the recording devices used:\n";
for (my $i = 0; $i < $num_dev; $i++)
{
    $v = $rec->get_info(dev => $i, info => 'dev_hw_name');
    my $name = $v->value();
    print "  device #$i: $name\n";
}
print "\n";
	
# print name for every channel
print "infos about the channels:\n";
for (my $i = 0; $i < $num_ch; $i++)
{
    $v = $rec->get_info(ch => $i, info => 'ch_name');
    my $name = $v->value();
    $v = $rec->get_info(ch => $i, info => 'ch_unit');
    my $unit = $v->value();
    print "  channel #$i: $name [$unit]\n";
}
print "\n";

exit 0;
#]]>

File: examples/handle_recs.py
#<![CDATA[
import sys
from RASCH import *

# initialize libRASCH
ra = RASCH()
if not ra:
    print "can't initialize libRASCH"
    sys.exit()

# open measurement
meas = ra.open_meas(sys.argv[1], '', 0)
if not meas:
    print "can't open measurement", sys.argv[1]
    sys.exit()

# get root recording
rec = meas.get_first_rec(0)
if not rec:
    print "can't get root recording"
    sys.exit()

# get some infos about recording
num_dev = rec.get_info(info='rec_num_devices')
num_ch = rec.get_info(info='rec_num_channel')
rec_name = rec.get_info(info='rec_name')
rec_date = rec.get_info(info='rec_date')
print "measurement", rec_name.value()
print "recorded at", rec_date.value()
print "#devices =", num_dev.value()
print "#channels =", num_ch.value()
print

# print name for every device
print "infos about the recording devices used:"
if num_dev.value() > 0:
    for i in range(num_dev.value()):
        name = rec.get_info(dev=i, info='dev_hw_name')
        print "  device #%d: %s" % (i, name.value())
print ""
	
# print name for every channel
print "infos about the channels:";
if num_ch.value() > 0:
    for i in range(num_ch.value()):
        name = rec.get_info(ch=i, info='ch_name')
        unit = rec.get_info(ch=i, info='ch_unit')
        print "  channel #%d: %s [%s]" % (i, name.value(), unit.value())
print "";
    
#]]>
